### PR TITLE
[dev-bug] - A failure in pass_files_to_docs_review - release_notes_review step

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [ link to the issue](https://github.com/demisto/etc/issues/48126)

## Description
- Fixed an issue where doc-review failed to recognize scripts name from old versions (when no scripts dir exist).
- Fixed a bug in the non-used function `_create_yaml_based`. I used this function now as part of my UTs.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 
 * Added the flag '-d', '--dependency' to **find-dependencies** command to get the content items that cause the dependencies between two packs.
+* Fixed an issue where **doc-review** failed to recognize scripts' names from old structure.
```
Comment (dorschw) [inline CHANGELOG.md:6]:
```suggestion
* Fixed an issue where **doc-review** failed to recognize script names, in scripts using the old file structure.
```

File: TestSuite/pack.py [hunk 2]
```diff
@@ -280,7 +281,7 @@ def _create_yaml_based(
             content: dict = {},
     ) -> YAML:
         yaml_name = f"{name}.yml"
-        yaml_path = dir_path / yaml_name
+        yaml_path = Path(os.path.join(dir_path, yaml_name))
```
Comment (dorschw) [inline TestSuite/pack.py:284]:
```suggestion
        yaml_path = Path(dir_path, yaml_name)
```

File: TestSuite/pack.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import os.path
```
Comment (dorschw) [inline TestSuite/pack.py:1]:
```suggestion
```

File: demisto_sdk/commands/common/tools.py [hunk 4]
```diff
@@ -901,11 +901,19 @@ def get_scripts_names(file_path):
...
-            script_path_full = os.path.join(scripts_dir_path, script, f'{script}.yml')
-            yml_dict = get_yaml(script_path_full)
-            click.secho(f'name: {yml_dict.get("name")}')
-            scripts_names.add(yml_dict.get("name"))
-
+            if script.endswith('.md'):
+                continue  # in case the script is in the old version of CommonScripts - JS code, ignore the md file
+            elif script.endswith('.yml'):
+                # in case the script is in the old version of CommonScripts - JS code, only yml exists not in a dir
+                script_path_full = os.path.join(scripts_dir_path, script)
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:908]:
```suggestion
                script_path_full = str(Path(scripts_dir_path, script))
```

File: demisto_sdk/commands/common/tools.py [hunk 5]
```diff
@@ -901,11 +901,19 @@ def get_scripts_names(file_path):
...
+            else:
+                script_path_full = os.path.join(scripts_dir_path, script, f'{script}.yml')
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:910]:
```suggestion
                script_path_full = str(Path(scripts_dir_path, script, f'{script}.yml'))
```

File: demisto_sdk/commands/doc_reviewer/tests/doc_reviewer_test.py [hunk 6]
```diff
@@ -827,6 +828,33 @@ def test_find_known_words_from_pack_ignore_scripts_name(repo):
...
+    script1_name = 'script-first_script'
+    pack._create_yaml_based(name=script1_name, dir_path=f'{pack.path}//Scripts', content={'name': script1_name})
+    script2 = pack.create_script(name='second_script')
+    rn_file = pack.create_release_notes(version='1_0_0', content=f'{script1_name}\n{script2.name}')
+    doc_reviewer = DocReviewer(file_paths=[])
+
+    with ChangeCWD(repo.path):
+        found_known_words = doc_reviewer.find_known_words_from_pack(rn_file.path)[1]
+        assert script1_name in found_known_words
+        assert script2.name in found_known_words
```
Comment (dorschw) [inline demisto_sdk/commands/doc_reviewer/tests/doc_reviewer_test.py:855]:
```suggestion
    assert script1_name in found_known_words
    assert script2.name in found_known_words
```

File: demisto_sdk/commands/common/tools.py [hunk 7]
```diff
@@ -901,11 +901,19 @@ def get_scripts_names(file_path):
...
+            try:
+                yml_dict = get_yaml(script_path_full)
+                scripts_names.add(yml_dict.get("name"))
+            except FileNotFoundError:
```
Comment (dorschw) [inline demisto_sdk/commands/common/tools.py:914]: Does `get_yaml` throw this error? 
IIRC it does not (but return a `{}`, fix PR in the making..), please check with `if not <path object>.exists()`

**Review by dorschw — COMMENTED** [review]
nice work

File: CHANGELOG.md (hunk 1 above)
Comment (abaumgarten) [reply CHANGELOG.md:6]: Done

File: TestSuite/pack.py (hunk 3 above)
Comment (abaumgarten) [reply TestSuite/pack.py:1]: Done

File: TestSuite/pack.py (hunk 2 above)
Comment (abaumgarten) [reply TestSuite/pack.py:284]: Done

**Review by abaumgarten — COMMENTED** [review]
@dorschw thanks!

File: demisto_sdk/commands/doc_reviewer/tests/doc_reviewer_test.py (hunk 6 above)
Comment (DinaMeylakh) [inline demisto_sdk/commands/doc_reviewer/tests/doc_reviewer_test.py:855]: lets choose one either script_name or script.name

File: demisto_sdk/commands/common/tools.py [hunk 8]
```diff
@@ -901,11 +901,19 @@ def get_scripts_names(file_path):
         click.secho(f'no scripts found')
     else:
         for script in found_scripts:
-            script_path_full = os.path.join(scripts_dir_path, script, f'{script}.yml')
-            yml_dict = get_yaml(script_path_full)
-            click.secho(f'name: {yml_dict.get("name")}')
-            scripts_names.add(yml_dict.get("name"))
-
+            if script.endswith('.md'):
+                continue  # in case the script is in the old version of CommonScripts - JS code, ignore the md file
```
Comment (DinaMeylakh) [inline demisto_sdk/commands/common/tools.py:905]: let's try not to use `continue` in the code. If you insist, add a unittest to cover this line.

File: demisto_sdk/commands/common/tools.py [hunk 9]
```diff
@@ -901,11 +901,19 @@ def get_scripts_names(file_path):
...
+                # in case the script is in the old version of CommonScripts - JS code, only yml exists not in a dir
+                script_path_full = os.path.join(scripts_dir_path, script)
+            else:
+                script_path_full = os.path.join(scripts_dir_path, script, f'{script}.yml')
+            try:
+                yml_dict = get_yaml(script_path_full)
+                scripts_names.add(yml_dict.get("name"))
+            except FileNotFoundError:
+                # we couldn't load the script as the path is not fit Content convention scripts' names
+                scripts_names.add(script)
```
Comment (DinaMeylakh) [inline demisto_sdk/commands/common/tools.py:916]: Cover this with a unittest

File: demisto_sdk/commands/common/tools.py (hunk 8 above)
Comment (DinaMeylakh) [reply demisto_sdk/commands/common/tools.py:905]: Also, wouldn't we want to add the file name to the known words, even if it ends with .md ?

**Review by DinaMeylakh — CHANGES_REQUESTED** [review]
Looking good, see my comments and make sure you test with running locally.

File: demisto_sdk/commands/doc_reviewer/tests/doc_reviewer_test.py (hunk 6 above)
Comment (abaumgarten) [reply demisto_sdk/commands/doc_reviewer/tests/doc_reviewer_test.py:855]: This is not the same. Script1 has no name as an attribute. Script2 has name and it is better using it.

File: demisto_sdk/commands/common/tools.py (hunk 8 above)
Comment (abaumgarten) [reply demisto_sdk/commands/common/tools.py:905]: The only option to have .md in a scripts directory is when we have a old version (JS code). In that case, we take care of the yml file (in the next condition).  So, we would like to ignore .md of a script.

Comment (abaumgarten) [reply demisto_sdk/commands/common/tools.py:905]: An assert for the `continue` statement was added to the UT I added

File: demisto_sdk/commands/common/tools.py (hunk 9 above)
Comment (abaumgarten) [reply demisto_sdk/commands/common/tools.py:916]: I cannot create a wrong file (in order to have a UT) as our SDK test suit creates correct files form. I have tested it in a content draft PR and it is ok.

Comment (dorschw) [reply demisto_sdk/commands/common/tools.py:916]: We often create files using SDK/TestSuite, then modify them to make them invalid, can it be of use here?

**Review by DinaMeylakh — APPROVED** [review]
Nice work

**Pull request merged**